src/browserbox-imap.js

```diff
@@ -111,12 +117,7 @@
      */
     ImapClient.prototype.connect = function() {
         return new Promise((resolve, reject) => {
-            this.socket = this._TCPSocket.open(this.host, this.port, {
-                binaryType: 'arraybuffer',
-                useSecureTransport: this.secureMode,
-                ca: this.options.ca,
-                tlsWorkerPath: this.options.tlsWorkerPath
-            });
+            this.socket = this._TCPSocket.open(this.host, this.port, this.options.tcpSocket);
 
```

Comment:
It only solves the problem of setting the web socket options for the
wsShim. It doesn't provide a general approach to potentially set varying
options on all of the shims.

If you continue using the parameter mapping approach you will have to
modify ImapClient.prototype.connect any time an underlying shim changes. My
proposed solution is more generally applicable. If a new option becomes
available in the wsShim, winShim, nodeShim, or chromeShim it will be
accessible to the caller. They can add any TCPSocket Shim specific property
to the options.tcpSocket object.



On Thu, Dec 10, 2015 at 8:26 AM, Felix Hammerl <notifications@github.com>
wrote:

> In src/browserbox-imap.js
> <https://github.com/whiteout-io/browserbox/pull/74#discussion_r47224971>:
>
> > @@ -111,12 +117,7 @@
> >       */
> >      ImapClient.prototype.connect = function() {
> >          return new Promise((resolve, reject) => {
> > -            this.socket = this._TCPSocket.open(this.host, this.port, {
> > -                binaryType: 'arraybuffer',
> > -                useSecureTransport: this.secureMode,
> > -                ca: this.options.ca,
> > -                tlsWorkerPath: this.options.tlsWorkerPath
> > -            });
> > +            this.socket = this._TCPSocket.open(this.host, this.port, this.options.tcpSocket);
> >
>
> Wouldn't this be easier?
>
> this.socket = this._TCPSocket.open(this.host, this.port, {
>     binaryType: 'arraybuffer',
>     useSecureTransport: this.secureMode,
>     ca: this.options.ca,
>     tlsWorkerPath: this.options.tlsWorkerPath,
>     ws: this.options.ws
> });
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/whiteout-io/browserbox/pull/74/files#r47224971>.
>



-- 
Darrel O'Pry
The Spry Group, LLC.
http://www.spry-group.com
718-355-9767 x101